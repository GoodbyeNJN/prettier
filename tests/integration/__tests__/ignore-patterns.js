describe("ignore patterns from cli arguments should work", () => {
  runCli("cli/ignore-patterns/cli-arguments", [
    ".",
    "--ignore-pattern=*.ignored.js",
    "-l",
  ]).test({
    status: 1,
    stderr: "",
    write: [],
  });
});

describe("ignore patterns from config options should work", () => {
  runCli("cli/ignore-patterns/config-options", [".", "-l"]).test({
    status: 1,
    stderr: "",
    write: [],
  });
});

describe("ignore patterns from cli arguments should overrides config options", () => {
  runCli("cli/ignore-patterns/cli-overrides-config", [
    ".",
    "-l",
    "--ignore-pattern=*.ignored.js",
  ]).test({
    status: 1,
    stderr: "",
    write: [],
  });
});

describe("ignore patterns from cli or config should merge with .prettierignore and .gitignore", () => {
  runCli("cli/ignore-patterns/merge-dot-ignore", [".", "-l"]).test({
    status: 1,
    stderr: "",
    write: [],
  });
  runCli("cli/ignore-patterns/merge-dot-ignore", [
    ".",
    "-l",
    "--ignore-pattern=cli.ignored.js",
  ]).test({
    status: 1,
    stderr: "",
    write: [],
  });
  runCli("cli/ignore-patterns/merge-dot-ignore", [
    ".",
    "-l",
    "--ignore-path=gitignore",
  ]).test({
    status: 1,
    stderr: "",
    write: [],
  });
  runCli("cli/ignore-patterns/merge-dot-ignore", [
    ".",
    "-l",
    "--ignore-path=gitignore",
    "--ignore-pattern=cli.ignored.js",
  ]).test({
    status: 1,
    stderr: "",
    write: [],
  });
});

describe("ignore patterns from config in sub-directories should not be applied", () => {
  runCli("cli/ignore-patterns/sub-directories", [".", "-l"]).test({
    status: 1,
    stderr: "",
    write: [],
  });
  runCli("cli/ignore-patterns/sub-directories", ["sub", "-l"]).test({
    status: 1,
    stderr: "",
    write: [],
  });
  runCli("cli/ignore-patterns/sub-directories", ["sub/file.js", "-l"]).test({
    status: 1,
    stderr: "",
    write: [],
  });
});

describe("overrides option in config should not have ignorePatterns", () => {
  runCli("cli/ignore-patterns/config-options-overrides", [".", "-l"]).test({
    status: 1,
    stderr: "",
    write: [],
  });
});
