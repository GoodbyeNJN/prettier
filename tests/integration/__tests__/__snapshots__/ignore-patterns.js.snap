// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignore patterns from cli arguments should overrides config options (stdout) 1`] = `"file.js"`;

exports[`ignore patterns from cli arguments should work (stdout) 1`] = `"file.js"`;

exports[`ignore patterns from cli or config should merge with .prettierignore and .gitignore (stdout) 1`] = `
"cli.ignored.js
file.js
git.ignored.js"
`;

exports[`ignore patterns from cli or config should merge with .prettierignore and .gitignore (stdout) 2`] = `
"config.ignored.js
file.js
git.ignored.js"
`;

exports[`ignore patterns from cli or config should merge with .prettierignore and .gitignore (stdout) 3`] = `
"cli.ignored.js
file.js
prettier.ignored.js"
`;

exports[`ignore patterns from cli or config should merge with .prettierignore and .gitignore (stdout) 4`] = `
"config.ignored.js
file.js
prettier.ignored.js"
`;

exports[`ignore patterns from config in sub-directories should not be applied (stdout) 1`] = `"sub/file.js"`;

exports[`ignore patterns from config in sub-directories should not be applied (stdout) 2`] = `"sub/file.js"`;

exports[`ignore patterns from config in sub-directories should not be applied (stdout) 3`] = `"sub/file.js"`;

exports[`ignore patterns from config options should work (stdout) 1`] = `"file.js"`;

exports[`overrides option in config should not have ignorePatterns (stdout) 1`] = `"file.js"`;
